### PR TITLE
Slack Signature Verification

### PR DESCRIPTION
**Pull request by sean-brydon**
## What does this PR do?

Verifies the signature with the incoming request from slack. This ensures that the request is actually coming from slack. 

## Type of change
- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?
- [ ] Preform a request to slack /create-event. Use a tool `ngrok` to monitor the request 
- [ ] Copy request and send it yourself with a modified signature.
- [ ] Request should fail

## Checklist
- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/app-store/slackmessaging/lib/slackVerify.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  if (!signingSecret) {
+    return res.status(400).json({ message: "Missing process.env.SLACK_SIGNING_SECRET" });
+  }
+
+  if (Math.abs(currentTime - parseInt(timeStamp)) > 60 * 5) {
+    return res.status(400).json({ message: "Request is too old" });
+  }
+
+  const signature_base = `v0:${timeStamp}:${stringify(body)}`;
+  const signed_sig = "v0=" + createHmac("sha256", signingSecret).update(signature_base).digest("hex");
```
Comment (sean-brydon) [inline packages/app-store/slackmessaging/lib/slackVerify.ts:27]: https://api.slack.com/authentication/verifying-requests-from-slack

File: packages/app-store/slackmessaging/api/interactiveHandler.ts [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
 import createEvent from "../lib/actions/createEvent";
+import slackVerify from "../lib/slackVerify";
 
 enum InteractionEvents {
   CREATE_EVENT = "cal.event.create",
 }
 
 export default async function interactiveHandler(req: NextApiRequest, res: NextApiResponse) {
   if (req.method === "POST") {
+    slackVerify(req, res);
```
Comment (emrysal) [inline packages/app-store/slackmessaging/api/interactiveHandler.ts:12]: Should this be `await slackVerify(...`?

Comment (sean-brydon) [reply packages/app-store/slackmessaging/api/interactiveHandler.ts:12]: You're right! Will make the change idk how I missed that

**Review by emrysal — APPROVED** [review]

**Review by zomars — APPROVED** [review]

**Review by agustif — APPROVED** [review]
Great work as always @sean-brydon, Code looks good to me!

File: packages/app-store/slackmessaging/lib/slackVerify.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { createHmac } from "crypto";
+import dayjs from "dayjs";
+import { NextApiRequest, NextApiResponse } from "next";
+import { stringify } from "querystring";
+
+const signingSecret = process.env.SLACK_SIGNING_SECRET;
```
Comment (zomars) [inline packages/app-store/slackmessaging/lib/slackVerify.ts:6]: This lives on the DB now @sean-brydon

**Pull request merged**